linked_list_mod.f90

```diff
@@ -94,6 +97,7 @@ subroutine append(this, value)
     else
        this%tail%next => node_ptr
        this%tail      => node_ptr
+       node_ptr%prev => this%tail
```

Comment:
Shouldn't this line be moved one line up? Otherwise `node_ptr%prev` just points at `node_ptr`.